pyccl/halos/halo_model.py

```diff
@@ -309,18 +356,27 @@ def I_1_2(self, cosmo, k, a, prof, *, prof2=None, prof_2pt):
                 a profile covariance object
                 returning the the two-point moment of the two profiles
                 being correlated.
+            diag (bool): If True, both halo profiles depend on the same k. If
+            False, they will depend on k and k', respectively. Default True.
```

Comment:
Can you check if this reads well in the doc pages?
Also, is the idea that, if diag = False, then k is a 2D array, not a 1D array?

Reply:
Checked and fixed. Yes `diag=False` returns a 2D array (k, k')